local-interchain/interchain/handlers/container_log_stream.go

```diff
@@ -0,0 +1,153 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"unicode"
+
+	dockertypes "github.com/docker/docker/api/types"
+	dockerclient "github.com/docker/docker/client"
+	"github.com/strangelove-ventures/interchaintest/v8/chain/cosmos"
+	"go.uber.org/zap"
+)
+
+var removeColorRegex = regexp.MustCompile("\x1b\\[[0-9;]*m")
+
+type ContainerStream struct {
+	ctx      context.Context
```

Comment:
Is there a specific reason we have to use a Context as a field member of this struct vs. piping a Context into the `StreamContainer` method or wherever a Context is needed for the `ContainerStream`?

It's typically more [idiomatic to pass a Context around vs. storing one inside of a struct type](https://go.dev/blog/context-and-structs)

One of the exceptions to this rule are typically when you have an external request that starts a background process/operation. 

Edit: After reading the rest of the code it seems like that is what may be happening here when a user makes a request to `/container_logs` so this may be a correct usage of context in a struct but figured I would ask for clarity.

Reply:
tldr; better Dev UX for the same relative solution

It's cleaner than having to write methods return `func(w http.ResponseWriter, req *http.Request) ` signatures just for ctx to be passed in in a wrapped type.

Current:
```go
r.HandleFunc("/container_logs", containerStream.StreamContainer)
```

Would be required change
```go
r.HandleFunc("/container_logs", containerStream.handleStreamContainer(ctx))

func (cs *ContainerStream) handleStreamContainer(ctx context) func(w http.ResponseWriter, req *http.Request) {

   return func(w http.ResponseWriter, req *http.Request) {
      // all the same logic here, just wrapped now
    }
}
```

ctx is only used for timeout limits and set on startup. So any runtime changes are unaffected in either approach - and the first approach is just net easier for a new contributor to wrap their head around.